app/src/main/java/co/icanteach/projectx/common/Extensions.kt

```diff
@@ -50,10 +53,5 @@ fun <T> Observable<Resource<T>>.doOnSuccess(
 
 }
 
-operator fun CompositeDisposable.plusAssign(disposable: Disposable) {
-    add(disposable)
-}
-
-fun Any?.runIfNull(block: () -> Unit) {
-    if (this == null) block()
-}
+fun Disposable.addTo(compositeDisposable: CompositeDisposable): Disposable =
+    apply { compositeDisposable.add(this) }
```

Comment:
you may change this  ` apply(compositeDisposable::add)` and  remove `: Disposable`